src/main/java/nextstep/subway/entity/Sections.java

```diff
@@ -75,15 +76,55 @@ public List<Station> getAllStations() {
         return new ArrayList<>(allStations);
     }
 
-    public boolean isConnectToLastStation(Section toSaveSection) {
-        return findLastStation().equals(toSaveSection.getUpStation());
+    public boolean hasExistingStation(Station station) {
+        return sections.stream()
+                .anyMatch(section -> section.isLeastOneSameStation(station));
     }
 
     public boolean hasNoSections() {
         return sections.isEmpty();
     }
 
+    private boolean isAddFirstSection(Section section) {
+        return sections.isEmpty();
+    }
+
+    private void insertSection(Section sectionToAdd) {
+        int insertionIndex = findInsertionIndex(sectionToAdd);
+        Section currentSection = sections.get(insertionIndex);
+
+        sections.set(insertionIndex, sectionToAdd);
+        sections.add(insertionIndex + 1, createNextSection(sectionToAdd, currentSection));
+    }
+
+    private Section createNextSection(Section sectionToAdd, Section sectionOfIndex) {
+        return new Section(
+                sectionToAdd.getDownStation(),
+                sectionOfIndex.getDownStation(),
+                sectionOfIndex.getDistance() - sectionToAdd.getDistance());
+    }
+
+    private int findInsertionIndex(Section sectionToAdd) {
+        return IntStream.range(0, sections.size())
```

Comment:
```suggestion
        return sections.stream()
```
으로 쓰는게 더 직관적이지 않나요? 🤔 
다른곳에서는 이렇게 쓰신것 같아요 😢

Reply:
👍 

궁금한 부분이 있는데요. 
인덱스를 반환하더라도 `IntStream` 사용은 자제하는 것이 좋을까요?

Reply:
사실 꼭 그렇지는 않은데, 반드시 인덱스를 반환해야 하는 경우가 얼마나 있을까 싶어요 🤔 
어쨋든 특정 객체를 찾고 싶은 건데, 꼭 인덱스를 찾는 경우가 많을까 싶기도 하고, IntStream 을 사용할 일이 많지는 않은 것 같아서 가독성이 떨어진다고 생각했습니다 😅